lib/parser/source/tree_rewriter.rb

```diff
@@ -326,6 +318,24 @@ def transaction
         @in_transaction = previous
       end
 
+      ##
+      # Provides access to a diagnostic engine.
+      # By default outputs diagnostic to $stderr
+      #
+      def self.default_diagnostics
```

Comment:
It is supposed to be static, right? Then maybe let's turn it into a constant? Or do you want it to be override-able on the subclass level?

Also, WDYT about moving it under `Diagnostic::Engine` namespace? `Diagnostic::Engine::Default` or `Diagnostic::Engine::Stderr`

Reply:
Yeah, I did think of moving it to the `Diagnostic:Engine`, but then I thought it could be nice to be overridable from subclass (`RuboCop` does subclass it), but it's not really that useful. A constant could work too.